components-api/src/main/scala/pl/touk/nussknacker/engine/api/process/Source.scala

```diff
@@ -49,6 +49,26 @@ trait TestWithParametersSupport[+T] { self: Source =>
   def parametersToTestData(params: Map[ParameterName, AnyRef]): T
 }
 
+/**
+ * Optional functionality which should provide field definitions based on input schema
+ * Based on those fields UI creates a window allowing user to test scenario based on schema.
+ */
+// This trait extends TestWithParametersSupport in order to preserve binary compatibility.
```

Comment:
Ignore binary compatibility and change it in this change. We can discuss why this is better approach, but I'd rather prefer to do this offline, because it is a bit longer discussion